storage/series.go

```diff
@@ -217,3 +256,33 @@ type errChunksIterator struct {
 func (e errChunksIterator) At() chunks.Meta { return chunks.Meta{} }
 func (e errChunksIterator) Next() bool      { return false }
 func (e errChunksIterator) Err() error      { return e.err }
+
+// ExpandSamples iterates over all samples in the iterator, buffering all in slice.
+// Optionally it takes samples constructor, useful when you want to compare sample slices with different
```

Comment:
This is a bit of an ugly way to have an optional parameter.

Reply:
Agreed. Better to use a normal parameter that can be `nil` if you want the default behavior.

Reply:
ack (: